CHANGELOG.md

```diff
@@ -1,5 +1,141 @@
+  (#2493) by @seongminn
+- **Breaking change:** Support `initialFocus` and `finalFocus` functions.
+  The `initialFocus` and `finalFocus` props can be functions that return DOM elements to focus.
+  The element must be returned directly (not as a ref).
+  (#2536) by @atomiks
+- Fix menu not opening when inside context menu trigger (#2506) by @baptisteArno
+- Fix `transform-origin` variable calculation when Positioner `sideOffset` is a function (#2511) by @atomiks
+- Fix submenu events (#2483) by @atomiks
+- Fix limitShift offset based on arrow size (#2571) by @atomiks
+
+### Navigation Menu
+
+- **Breaking change:** Semantic element structure and `active` page prop.
+  `NavigationMenu.List` renders `<ul>` and `NavigationMenu.Item` renders `<li>` by default.
+  (#2526) by @atomiks
+- Unshare AbortController instance (#2441) by @tomokinat
+- Close on link click by default (#2535) by @atomiks
+
+### Number Field
+
+- Fix duplicate `onValueChange` calls (#2591) by @atomiks
+
+### Popover
+
+- **Breaking change:** Support `initialFocus` and `finalFocus` functions.
+  The `initialFocus` and `finalFocus` props can be functions that return DOM elements to focus.
+  The element must be returned directly (not as a ref).
+  (#2536) by @atomiks
+- Fix outside click after right clicking in popup (#2508) by @baptisteArno
+- Fix unexpected close when nested inside two popovers (#2481) by @atomiks
+- Fix `transform-origin` variable calculation when Positioner `sideOffset` is a function (#2511) by @atomiks
+- Restore focus to popup when focused element is removed (#2479) by @atomiks
+- Fix limitShift offset based on arrow size (#2571) by @atomiks
+
+### Preview Card
+
+- Fix `transform-origin` variable calculation when Positioner `sideOffset` is a function (#2511) by @atomiks
+- Fix limitShift offset based on arrow size (#2571) by @atomiks
+
+### Radio Group
+
+- Return null in form data when no option selected (#2473) by @ingokpp
+
+### Scroll Area
+
+- Prevent pointer events from sibling portals triggering hover (#2542) by @KenanYusuf
+
+### Select
+
+- Fix stale `items` prop (#2397) by @atomiks
+- Fix unexpected close when nested inside two popovers (#2481) by @atomiks
+- Fix `onValueChange` type inference (#2372) by @atomiks
+- Fix `transform-origin` variable calculation when Positioner `sideOffset` is a function (#2511) by @atomiks
+- Reset state when selected item is removed (#2577) by @atomiks
+- Fix `data-highlighted` and DOM focus item desync (#2569) by @atomiks
+- Fix item click with `defaultOpen` prop (#2570) by @atomiks
+- Fix scroll arrows not propagating scroll fully to start/end of list (#2523) by @atomiks
+- Fix limitShift offset based on arrow size (#2571) by @atomiks
+
+### Slider
+
+- **Breaking change:** The `input type="range"` element receives focus and the thumb div is no longer focusable, focus styles targeting the thumb.
+  For example `.Thumb:focus-visible` should use `.Thumb:has(:focus-visible)` instead.
```

Comment:
```suggestion
- **Breaking change:** Instead of the thumb div, the `input type="range"` element receives focus. Focus styles that were targeting the thumb, should be updated.
  For example `.Thumb:focus-visible` should be replaced with `.Thumb:has(:focus-visible)`.
```